### PR TITLE
feat(ENG-860): Adapt the dev filter to handle teams once created

### PR DESCRIPTION
**Pull request by navassouza**
Closes: https://athenianco.atlassian.net/browse/ENG-860

Filter by teams with Option groups

### CODE REVIEW

**Comment by se7entyse7en** [comment]
@navassouza is this ready for review?

**Comment by navassouza** [comment]
@se7entyse7en, not yet. Just fixed those re-renders, they were cause the filter to halt for a second. I'll commit what I have so far and will continue

**Comment by se7entyse7en** [comment]
@warenlg I don't know whether the feature is expected to behave as-is, but testing this locally there's a specific case that felt weird to me.
If I have a member X in both Team A and Team B, and I deselect the whole Team A, it happens that also Team B becomes not selected given that member X has been deselected as well from Team B. IMO it's counter-intuitive and I even thought there was a bug until I understood what was the rationale.
I don't know though if and how you would solve this.

File: src/js/context/Data.jsx [hunk 1]
```diff
@@ -1,62 +1,73 @@
...
     if (global) {
-        if (!globalKeysWhitelist.includes(id)) {
-            throw Error(`Trying to use unrecognized global id: ${id}`);
-        } else if (newState.data[id]) {
-            throw Error(`Trying to override global data with id: ${id}`);
-        }
-    } else if (id.startsWith(globalKeyPrefix)) {
-        throw Error(`Non-global data cannot have id with prefix ${globalKeyPrefix}: ${id}`);
+      // 'filter.teams is whitelabled but not obrigatory
+      if (![...globalKeysWhitelist, 'filter.teams'].includes(key)) {
```
[the remark this answers is not in the document]
Comment (navassouza) [reply src/js/context/Data.jsx:38]: One problem I see is for any small change in the interface we hammer all API endpoints, but in the case of teams, I'm not sure if we should pull it all the time, I'm considering caching it and when we call `resetGlobal()` we skip teams from being removed. So we have 1 less request being called over and over. What you think?

File: src/js/context/Data.jsx [hunk 2]
```diff
@@ -1,62 +1,73 @@
...
+    } else if (key.startsWith(globalKeyPrefix)) {
+      throw Error(`Non-global data cannot have id with prefix ${globalKeyPrefix}: ${key}`)
     }
+    
+    const prefix = global ? globalKeyPrefix : ''
+    // if 'id' is an array of whitelabeled keys, get data for each id (line:33 commment)
+    reducedData.data[`${prefix}${key}`] = Array.isArray(id) ? dataArr[index] : data
+  
+    // filter.teams is skipped below
+    reducedData.globalReady = _(globalKeysWhitelist).map(k => !!reducedData.data[`global.${k}`]).every()
```
[the remark this answers is not in the document]
Comment (navassouza) [reply src/js/context/Data.jsx:52]: We'll treat teams just like any other data source for now.

**Comment by warenlg** [comment]
Not sure if it was clear enough during the syn. Just in case to summarize on Marvin's comment:
* we remove the product requirement saying that all the checkboxes of one same dev should be consistent. We allow a dev to be checked in one team and unchecked in another.
* If a dev is selected in at least one team, then it is selected from the API point of view.

File: src/js/context/Data.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (se7entyse7en) [reply src/js/context/Data.jsx:38]: discussed in Slack: let's uniform the behaviour and iterate on eventually improving it in a second moment.

**Comment by navassouza** [comment]
@warenlg 

> A dev could be assigned to several teams. In that case, Frontend should ensure that all checkboxes of one dev are either ticked or unticked but we should prevent the UI to show one ticked and the other unticked.

from https://athenianco.atlassian.net/browse/ENG-726

I understood was, if a dev is selected, it's selected/unselected everywhere

**Comment by warenlg** [comment]
Yes that's why I say now that we have to remove this product requirement in order to satisfy Marvin's comment and the concerns expressed during the sync today.

> A dev could be assigned to several teams. In that case, Frontend should ensure that all checkboxes of one dev are either ticked or unticked but we should prevent the UI to show one ticked and the other unticked.
We remove this requirement.

**Comment by se7entyse7en** [comment]
@navassouza once what we discussed is done feel free to merge this 👍

**Comment by se7entyse7en** [comment]
We're still storing already _awaited_ objects in the store that means it will make charts wait for them.

**Pull request merged**